File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l=0,h=nums.size()-1;
        while(l<=h){
            int mid=l+(h-l)/2;
            if(nums[mid]>=nums[0] && nums[0]>nums[nums.size()-1]) l=mid+1;
            else h=mid-1;
        }
        return nums[l];
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int n = nums.length;
        int l =0,h=n-1;
        while(l<=h) {
            int mid = l+(h-l)/2;
            if(nums[mid]>=nums[0] && nums[mid]>nums[n-1]) l=mid+1;
            else h=mid-1;
        }
        return nums[l];
    }
}